cpp/log/etcd_consistent_store-inl.h

```diff
@@ -314,6 +384,77 @@ std::string EtcdConsistentStore<Logged>::GetFullPath(
   return root_ + key;
 }
 
+
+// static
+template <class Logged>
+template <class T>
+Update<T> EtcdConsistentStore<Logged>::TypedUpdateFromWatchUpdate(
+    const EtcdClient::WatchUpdate& update) {
+  const std::string raw_value(util::FromBase64(update.node_.value_.c_str()));
+  T thing;
+  CHECK(thing.ParseFromString(raw_value)) << raw_value;
+  EntryHandle<T> handle(thing);
+  if (update.exists_) {
+    handle.SetHandle(update.node_.modified_index_);
+  }
+  return Update<T>(handle, update.exists_);
+}
+
+
+template <class Logged>
+void EtcdConsistentStore<Logged>::UpdateLocalServingSTH(
+    const std::unique_lock<std::mutex>& lock,
+    const EntryHandle<ct::SignedTreeHead>& handle) {
+  CHECK(lock.owns_lock());
+  CHECK(!serving_sth_ ||
+        serving_sth_->Entry().timestamp() < handle.Entry().timestamp());
+
+  VLOG(1) << "Updating serving_sth_ to: " << handle.Entry().DebugString();
+  serving_sth_.reset(new EntryHandle<ct::SignedTreeHead>(handle));
+}
+
+
+template <class Logged>
+void EtcdConsistentStore<Logged>::OnEtcdServingSTHUpdated(
+    const std::vector<EtcdClient::WatchUpdate>& updates) {
+  CHECK_LE(updates.size(), 1);
+
+  if (!updates.empty()) {
+    const Update<ct::SignedTreeHead> update(
+        TypedUpdateFromWatchUpdate<ct::SignedTreeHead>(updates[0]));
+    {
+      std::unique_lock<std::mutex> lock(mutex_);
+      UpdateLocalServingSTH(lock, update.handle_);
+    }
+    this->OnServingSTHUpdate(update);
```

Comment:
Now that both ```UpdateLocalServingSTH``` and ```OnServingSTHUpdate``` are called from just one place, it would make some sense to inline some of it? ```OnServingSTHUpdate``` is very short, for example...

Also, since you take the lock *just* for ```UpdateLocalServingSTH```, it might as well use a ```lock_guard``` in it?